### PR TITLE
refactor(ui): split check creation flow

### LINKED ISSUE #15162: As a user I can create a threshold check from the Monitoring & Alerting index
This action should open the check editor with the check type already set to `threshold`

The query builder should display a aggregate function selector

### LINKED ISSUE #15163: As a user I can create a deadman check from the Monitoring & Alerting index view
This should open the check editor with the type already set to `deadman`

The query builder should not display the aggregate function selector

### LINKED ISSUE #15164: Remove deadman/threshold toggle from check builder
Contingent upon #15162 & #15163

### LINKED ISSUE #15165: Replace create check button with dropdown
Dropdown should have 2 options:

`Threshold Check`
`Deadman Check`

### PR DESCRIPTION
**Pull request by alexpaxton**
Closes #15162 
Closes #15163 
Closes #15164 
Closes #15165 

Screen Shot 2019-09-16 at 5 11 12 PM
Have the user select the type of check they are making before opening the check editor

Screen Shot 2019-09-16 at 4 23 09 PM
Screen Shot 2019-09-16 at 4 23 16 PM
Move validation popover to save button instead of radio buttons

Screen Shot 2019-09-16 at 4 23 24 PM
Don't show function selector if check is deadman

Screen Shot 2019-09-16 at 4 23 52 PM
Remove check type toggle from conditions card

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/timeMachine/utils/queryBuilder.ts [hunk 1]
```diff
@@ -47,14 +47,20 @@ export const isDraftQueryAlertable = (
 }
 
 export const isCheckSaveable = (
```
Comment (ebb-tide) [inline ui/src/timeMachine/utils/queryBuilder.ts]: hrmm. I wonder if there are other things we should check for here? Maybe for another PR.

File: ui/src/alerting/components/CreateCheckDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+// Components
+import {Dropdown, ComponentColor, IconFont} from '@influxdata/clockface'
+
+interface Props {
+  onCreateThreshold: () => void
+  onCreateDeadman: () => void
+}
+
+enum CheckType {
```
Comment (ebb-tide) [inline ui/src/alerting/components/CreateCheckDropdown.tsx]: this type already exists in `/src/types/alerting`

File: ui/src/alerting/components/CreateCheckDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      testID="create-check"
+    >
+      Create
+    </Dropdown.Button>
+  )
+
+  const DropdownMenu = (onCollapse: () => void) => (
+    <Dropdown.Menu onCollapse={onCollapse}>
+      <Dropdown.Item
+        value={CheckType.Threshold}
```
Comment (ebb-tide) [inline ui/src/alerting/components/CreateCheckDropdown.tsx]: and here (and all the other places) you can just enter the string 'threshold' when you use the type from types

File: ui/src/alerting/components/builder/CheckConditionsCard.tsx [hunk 4]
```diff
@@ -26,60 +23,35 @@ interface StateProps {
...
-}
+type Props = StateProps
+
+const CheckConditionsCard: FC<Props> = ({check}) => {
+  let cardTitle: string
+  let conditionsComponent: JSX.Element
 
-type Props = DispatchProps & StateProps
+  if (check.type === 'deadman') {
+    cardTitle = 'Deadman'
```
Comment (ebb-tide) [inline ui/src/alerting/components/builder/CheckConditionsCard.tsx]: I wonder if this should be Deadman Conditions?

Comment (alexpaxton) [reply ui/src/alerting/components/builder/CheckConditionsCard.tsx]: Ahh like split it into two components entirely?

Comment (ebb-tide) [reply ui/src/alerting/components/builder/CheckConditionsCard.tsx]: I meant should the cardTitle be "Deadman Conditions"

**Pull request merged**